### PR TITLE
Update docs to cover recent updates

### PR DESCRIPTION
**Pull request by bwbarrett**
Add details covering hwloc, changes to our release process, and the use of dlopen to the README.

This is blocking on https://github.com/aws/aws-ofi-nccl/pull/224, so that it becomes true.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,9 +26,15 @@ The plug-in currently supports the following distributions:
...
+It requires [Libfabric](http://github.com/ofiwg/libfabric/),
+[NCCL](https://github.com/NVIDIA/nccl),
+[HWLOC](https://www.open-mpi.org/projects/hwloc/), and (if you want to
+run tests) an MPI Implementation. Please see the
 [Release notes](http://github.com/aws/aws-ofi-nccl/releases) for
-information on version compatibility.
+information on version compatibility.  We recommend using the
+distribution version of hwloc, which can be installed with `yum
+install hwloc-devel` on many RPM based distributions and `apt install
+libhwloc-dev` on many DPKG based distibutions.
```
Comment (AvivBenchorin) [inline README.md:37]: Does the latest version of the plugin just require the `hwloc-devel`/`libhwloc-dev` package, or also the `hwloc-plugins`/`libhwloc-plugins` package?

File: README.md [hunk 2]
```diff
@@ -111,6 +124,15 @@ config option:
 
 ### Plugin Configurations
 
+Similar to NCCL or Libfabric, the plugin dynamically loads CUDA
+dependencies at runtime, specifically `libcudart.so`.  Like NCL and
```
Comment (AmedeoSapio) [inline README.md:128]: NCL->NCCL

File: README.md (hunk 1 above)
Comment (bwbarrett) [reply README.md:37]: It does not require the plugins package.  Just hwloc.

File: README.md (hunk 2 above)
Comment (bwbarrett) [reply README.md]: blah; fixed.

**Review by AvivBenchorin — APPROVED** [review]

**Pull request merged**